src/WebApi/UnitTests/Tests/Application/UseCase/QuestionListTests/CreateQuestionListUseCaseTest.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Application.Repositories;
+using Application.UseCases.QuestionList.CreateQuestionList;
+using Domain.Models;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace UnitTests.Tests.Application.UseCase.QuestionListTests;
+
+public class CreateQuestionListUseCaseTest
+{
+    private static CreateQuestionListInput Input => new CreateQuestionListInput()
+    {
+        Title = "test",
+        Description = "test"
+    };
+
+    [Fact]
+    public async Task Execute_RepositoryNull_Throws()
+    {
+        var useCase = new CreateQuestionListUseCase(null);
+
+        var execute = async () => await useCase.Execute(Input);
+
+        await Assert.ThrowsAsync<NullReferenceException>(execute);
+    }
+
+    [Fact]
+    public async Task Execute_PassNull_Throws()
+    {
+        var useCase = new CreateQuestionListUseCase(It.IsAny<IQuestionListRepository>());
+
+        var execute = async () => await useCase.Execute(null);
+
+        await Assert.ThrowsAsync<ArgumentNullException>(execute);
+    }
+
+    [Fact]
+    public async Task Execute_PassValidInput_CallsOk()
+    {
+        var repositoryMock = new Mock<IQuestionListRepository>();
+        repositoryMock.Setup(x => x.Create(It.IsAny<QuestionListModel>()).Result).Returns(new QuestionListModel()
+        {
+            Title = "test",
+            Description = "test"
+        });
+        repositoryMock.Setup(x => x.AddQuestionsToList(It.IsAny<QuestionListModel>(), It.IsAny<IEnumerable<int>>()).Result)
+                      .Returns(true);
```

Comment:
Just a suggestion, let's hide these mocks into methods. After all, it's important to know what the test does, not really HOW it does it. So let's make the API a little better so that we can use something like:

```cs
ArrangeQuestionListCreate(new QuestionListModel
{
    Title = "test",
    Description = "test"
});
ArrangeAddQuestionsToListSuccess(true);
```

Again, feel free to name and describe them a lot better, and of course parameterize it if need be. Then we can even reuse those in other tests.

The point really being that the test has gotta be easy to read, otherwise people won't maintain it 🙂